combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/RepeatedProbeBleDevice.kt

```diff
@@ -230,39 +276,135 @@ internal class RepeatedProbeBleDevice (
-
-                    // Syncronous Requests that are responded to with a single message
-                    is NodeSetPredictionResponse -> setPredictionHandler.handled(message.success, null)
-                    is NodeReadFirmwareRevisionResponse -> probeFirmwareRevisionHandler.handled(message.success, message)
-                    is NodeReadHardwareRevisionResponse -> probeHardwareRevisionHandler.handled(message.success, message)
-                    is NodeReadModelInfoResponse -> probeModelInfoHandler.handled(message.success, message)
+                    // Unsupported Messages
+                    // is NodeSetColorResponse -> setColorHandler.handled(response.success, null)
+                    // is NodeSetIDResponse -> setIdHandler.handled(response.success, null)
+
+                    // Repeated Responses
+                    is NodeReadLogsResponse -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            routeMonitor.activity()
+                            handleLogResponse(message)
+                        }
+                    }
+
+                    // Synchronous Requests that are responded to with a single message
+                    is NodeSetPredictionResponse -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            routeMonitor.activity()
+                            setPredictionHandler.handled(message.success, null)
+                        }
+                    }
+                    is NodeReadFirmwareRevisionResponse -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            routeMonitor.activity()
+                            probeFirmwareRevisionHandler.handled(message.success, message)
+                        }
+                    }
+                    is NodeReadHardwareRevisionResponse -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            routeMonitor.activity()
+                            probeHardwareRevisionHandler.handled(message.success, message)
+                        }
+                    }
+                    is NodeReadModelInfoResponse -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            routeMonitor.activity()
+                            probeModelInfoHandler.handled(message.success, message)
+                        }
+                    }
 
                     /// Async Requests that are Broadcast on certain events from a Node
-                    is NodeProbeStatusRequest -> handleProbeStatusRequest(message)
-                    is NodeReadSessionInfoResponse -> sessionInfoHandler.handled(message.success, message.sessionInformation)
+                    is NodeProbeStatusRequest -> {
+                        if(message.serialNumber == probeSerialNumber) {
+                            handleProbeStatusRequest(message)
```

Comment:
Reading the description of `routeMonitor` above, it only calls out session info as the exception, but it sounds like it's the async events are really the differentiators?